tx.go

```diff
@@ -33,15 +33,20 @@ lock_time        if non-zero and sequence numbers are < 0xFFFFFFFF: block height
 --------------------------------------------------------
 */
 
+// Sentinel errors for transactions.
+var (
+	ErrInvalidTxID = errors.New("invalid TxID")
+)
+
 // Tx wraps a bitcoin transaction
 //
 // DO NOT CHANGE ORDER - Optimised memory via malign
 //
 type Tx struct {
-	Inputs   []*Input  `json:"vin"`
-	Outputs  []*Output `json:"vout"`
-	Version  uint32    `json:"version"`
-	LockTime uint32    `json:"locktime"`
+	inputs   []*Input
+	outputs  []*Output
```

Comment:
did we want to make the inputs and outputs fields private as well? might that not remove too much flexibility? we wouldn't be able to for example change an output satoshi amount anymore for example...

Reply:
Yeah we had originally talked about hiding both of these, perhaps only inputs should be internal, or shall I just make both public again? Idea was to reduce surface area of the api, but it could be annoying to use

Reply:
I've made both public again